server/http.go

```diff
@@ -235,6 +269,54 @@ func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
 	p.handleMeetingEnded(w, r, &meetingWebhook)
 }
 
+func (p *Plugin) handleParticipantJoinedLeftEvent(w http.ResponseWriter, event *zoom.ParticipantJoinedLeftEvent) {
+	participant := event.Payload.Object.Participant
+	userID, uerr := p.getMattermostUserIdFromZoomParticipant(participant)
+	if uerr != nil {
+		errMsg := "failed to get Mattermost user id for Zoom user"
+		p.API.LogWarn(errMsg, "err", uerr.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	followStatus, _ := p.getFollowStatusForUser(userID)
+	if followStatus == false {
+		return
+	}
+
+	currentStatus, csErr := p.API.GetUserStatus(userID)
+	if csErr != nil {
+		errMsg := "failed to get Mattermost user status"
+		p.API.LogWarn(errMsg, "err", csErr.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	p.API.LogDebug("Mattermost user status is " + currentStatus.Status)
+
+	var newStatus string
+	if event.EventType == zoom.EventTypeParticipantJoined &&
```

Comment:
0/5 should we consider whether the current status has been set manually or automatically?